forc-plugins/forc-client/src/op/deploy.rs

```diff
@@ -251,48 +276,174 @@ async fn deploy_new_proxy(
+        && predicates_to_deploy.is_empty()
+    {
+        println_warning("No deployable package is found in the current directory.");
+    } else {
+        deployed_packages.extend(
+            deploy_contracts(&command, &contracts_to_deploy)
+                .await?
+                .into_iter()
+                .map(DeployedPackage::Contract),
+        );
+        deployed_packages.extend(
+            deploy_executables(&command, &scripts_to_deploy)
+                .await?
+                .into_iter()
+                .map(DeployedPackage::Script),
+        );
+        deployed_packages.extend(
+            deploy_executables(&command, &predicates_to_deploy)
+                .await?
+                .into_iter()
+                .map(DeployedPackage::Predicate),
+        );
+    }
+
+    Ok(deployed_packages)
+}
+
+/// Builds and deploys executable (script and predicate) package(s) as blobs,
+/// and generates a loader for each of them.
+pub async fn deploy_executables(
+    command: &cmd::Deploy,
+    executables_to_deploy: &[Arc<BuiltPackage>],
+) -> Result<Vec<DeployedExecutable>> {
+    let mut deployed_executable = vec![];
+    if executables_to_deploy.is_empty() {
+        return Ok(deployed_executable);
+    }
+
+    let node_url = validate_and_get_node_url(command, executables_to_deploy).await?;
+    // We will have 1 transaction per executable as each deployment uses a single blob.
+    let tx_count = executables_to_deploy.len();
+    let account = setup_deployment_account(command, &node_url, tx_count).await?;
+
+    for pkg in executables_to_deploy {
+        let script = Executable::from_bytes(pkg.bytecode.bytes.clone());
+        let loader = script.convert_to_loader()?;
+        println_action_green("Uploading", "blob containing executable bytecode.");
+        loader.upload_blob(account.clone()).await?;
+        println_action_green("Generating", "loader bytecode for the uploaded executable.");
+        let loader_bytecode = loader.code();
+        let pkg_name = &pkg.descriptor.name;
+        let out_dir = pkg.descriptor.manifest_file.dir().join("out");
+        let bin_path = out_dir.join(format!("{pkg_name}-loader.bin"));
+        std::fs::write(&bin_path, &loader_bytecode)?;
+        println_action_green(
+            "Saved",
+            &format!("loader bytecode at {}", bin_path.display()),
+        );
+        // If the executable is a predicate, we also want to display and save the predicate root.
+        if pkg
+            .descriptor
+            .manifest_file
+            .program_type()
+            .with_context(|| {
+                "error while trying to retrieve program type for executable deployment."
+            })?
+            == TreeType::Predicate
+        {
+            // Calculate the root.
+            let root = format!("0x{}", fuel_tx::Input::predicate_owner(&loader_bytecode));
+            // Root files are named in `pkg-name-root` from, since this is a
```

Comment:
Nit: remove "from"

Reply:
can you make this a code suggestion so we can commit it directly

Reply:
oh i meant to make it `form`, thanks!